### DISCUSSION TITLE
Automate Installation of Python-lsp-server

### DISCUSSION THREAD

**Discussion by ebresie** [python-lsp/python-lsp-server#283] [Configuration]
I am working on a new LSP based Python plugin for Netbeans (see https://github.com/ebresie/python4nb) and leveraging the server for use in the plugin.

Presently setup is a manual setup with the call to "python -m pip install python-language-server[all]"

This work, but I'd like to "automate" the setup process.  I was thinking maybe some sort of setup.py or virtual environment with the given server installed but wasn't sure if there was any better way or guidance on how to do automate this.

**Comment by ebresie** [comment]
There may be some aspects of this in regard to documentation below, but this still focuses on manual install and startup and no so much automated setup.

1. https://github.com/python-lsp/python-lsp-server/issues/156
2. https://github.com/python-lsp/python-lsp-server/issues/157
3. https://github.com/python-lsp/python-lsp-server/pull/185

**Comment by doolio** [comment]
Maybe specify it as a dev package in a `pyproject.toml` file and have a tool like `poetry` install it automatically for you?
